contracts/src/interfaces/ITeleporterTokenBridge.sol

```diff
@@ -32,6 +32,75 @@ struct SendTokensInput {
     uint256 requiredGasLimit;
 }
 
+/**
+ * @notice Parameters for briding tokens to another chain and calling a contract on that chain
+ * @param destinationBlockchainID blockchainID of the destination
+ * @param destinationBridgeAddress address of the destination token bridge instance
+ * @param recipientContract the contract on the destination chain that will be called
+ * @param recipientPayload the payload that will be provided to the recipient contract on the destination chain
+ * @param requiredGasLimit the required amount of gas needed to deliver the message on its destination chain,
+ *                         including token operations and the call the recipient contract.
+ * @param recipientGasLimit the amount of gas that will provided to the recipient contract on the destination chain,
+ *                          which must be less than the requiredGasLimit of the message as a whole.
+ * @param fallbackRecipient address where the bridged tokens are sent if the call to the recipient contract fails.
+ * @param primaryFee amount of tokens to pay for Teleporter fee on the source chain
+ * @param secondaryFee amount of tokens to pay for Teleporter fee if a multihop is needed
+ */
+struct SendAndCallInput {
+    bytes32 destinationBlockchainID;
+    address destinationBridgeAddress;
+    address recipientContract;
+    bytes recipientPayload;
+    uint256 requiredGasLimit;
+    uint256 recipientGasLimit;
+    address fallbackRecipient;
+    uint256 primaryFee;
+    uint256 secondaryFee;
+}
+
+enum BridgeMessageType {
+    SINGLE_HOP_SEND,
+    SINGLE_HOP_CALL,
+    MULTI_HOP_SEND,
+    MULTI_HOP_CALL
+}
+
+struct BridgeMessage {
+    BridgeMessageType messageType;
+    uint256 amount;
+    bytes payload;
+}
+
+struct SingleHopSendMessage {
+    address recipient;
+}
+
+struct SingleHopCallMessage {
+    address recipientContract;
+    bytes recipientPayload;
+    uint256 recipientGasLimit;
+    address fallbackRecipient;
+}
+
+struct MultiHopSendMessage {
+    bytes32 destinationBlockchainID;
+    address destinationBridgeAddress;
+    address recipient;
+    uint256 secondaryFee;
+    uint256 secondaryGasLimit;
+}
+
+struct MultiHopCallMessage {
+    bytes32 destinationBlockchainID;
+    address destinationBridgeAddress;
+    address recipientContract;
+    bytes recipientPayload;
+    uint256 requiredGasLimit;
+    uint256 recipientGasLimit;
```

Comment:
I renamed it to `secondaryRequiredGasLimit` and added a doc comment 👍 We could wrap a `SingleHopCallMessage`, but I think it'd be a bit confusing to have a single hop embedded in a multi hop and it would make for long names when mapping a multihop message to a message input for the second `_sendAndCall` call.